### PR TITLE
Adding in timestamp and bar chart to temp monitor

### PR DESCRIPTION
**Pull request by ChangoFoster**
This change is [Reviewable](https://reviewable.io/reviews/aiky30/rpi-docs/2)

### CODE REVIEW

File: scripts/pi_temp.py [hunk 1]
```diff
@@ -1,12 +1,30 @@
 import os
 import time
+from datetime import datetime
 
+#Lists for graph
```
Comment (Aiky30) [inline scripts/pi_temp.py]: Comments should have a space between the # and the string. Change to:

`# Lists for graph`

File: scripts/pi_temp.py [hunk 2]
```diff
@@ -1,12 +1,30 @@
...
+x_axis = []
+y_axis = []
 
 def get_cpu_temp():
-        cpu_temp = os.popen("vcgencmd measure_temp").readline()
-        return cpu_temp.replace("temp=","")
-
+    cpu_temp = os.popen("vcgencmd measure_temp").readline()
+    #Convert the temp read from the OS to a clean float
```
Comment (Aiky30) [inline scripts/pi_temp.py]: Same comment as above regarding space

File: scripts/pi_temp.py (hunk 1 above)
Comment (Aiky30) [reply scripts/pi_temp.py]: This is the style guide for Python: https://www.python.org/dev/peps/pep-0008/

File: scripts/pi_temp.py [hunk 3]
```diff
@@ -1,12 +1,30 @@
...
-        print(get_cpu_temp())
-        time.sleep(.5)
+    #Add latest date
+    x_axis.append(get_time())
+    y_axis.append(get_cpu_temp())
+    #Create elements for graph
+    latest_time = str(x_axis[-1]) + " " 
+    latest_temp = str(y_axis[-1]) + " "
+    temp_lines = str("|" * (int(y_axis[-1]) - 30))
+    #Print line for graph every .5 seconds
```
Comment (Aiky30) [inline scripts/pi_temp.py]: Same as above regarding comment spacing

File: scripts/pi_temp.py [hunk 4]
```diff
@@ -1,12 +1,30 @@
...
 
+def get_time():
+    now = datetime.now()
+    current_time = now.strftime("%H:%M:%S") # 24-Hour:Minute:Second
+    return  current_time
+                        
 while True:
-        print(get_cpu_temp())
-        time.sleep(.5)
+    #Add latest date
```
Comment (Aiky30) [inline scripts/pi_temp.py]: Same as above regarding comment spacing

File: scripts/pi_temp.py [hunk 5]
```diff
@@ -1,12 +1,30 @@
...
+    x_axis.append(get_time())
+    y_axis.append(get_cpu_temp())
+    #Create elements for graph
```
Comment (Aiky30) [inline scripts/pi_temp.py]: Same as above regarding comment spacing

File: scripts/pi_temp.py [hunk 6]
```diff
@@ -1,12 +1,30 @@
...
+    latest_time = str(x_axis[-1]) + " " 
+    latest_temp = str(y_axis[-1]) + " "
+    temp_lines = str("|" * (int(y_axis[-1]) - 30))
+    #Print line for graph every .5 seconds
+    print(latest_time + latest_temp + temp_lines)
```
Comment (Aiky30) [inline scripts/pi_temp.py]: Python has a nice string format function that can be used many ways, the following way is how I tend to use it although there are many different styles on the web.

 I would do the above like so:
```python

temperature = get_cpu_temp()
bar_size = int(temperature) - 30
bar = str("|" * bar_size)

print("{time} {temp} {graph}".format(
    time=get_time(), temp=temperature, graph=bar
))
```

**Review by Aiky30 — CHANGES_REQUESTED** [review]
Left some comments regarding style changes and an alternative solution to the print.

File: scripts/pi_temp.py (hunk 1 above)
Comment (ChangoFoster) [reply scripts/pi_temp.py]: Updated in recent commit

File: scripts/pi_temp.py (hunk 3 above)
Comment (ChangoFoster) [reply scripts/pi_temp.py]: Updated in recent commit

File: scripts/pi_temp.py (hunk 5 above)
Comment (ChangoFoster) [reply scripts/pi_temp.py]: Updated in recent commit

File: scripts/pi_temp.py (hunk 6 above)
Comment (ChangoFoster) [reply scripts/pi_temp.py]: Updated in recent commit

File: scripts/pi_temp.py (hunk 4 above)
Comment (ChangoFoster) [reply scripts/pi_temp.py]: Updated in recent commit

File: scripts/pi_temp.py (hunk 2 above)
Comment (ChangoFoster) [reply scripts/pi_temp.py]: Updated in recent commit

**Comment by ChangoFoster** [comment]
Just one year later and I finally got around to addressing these!

**Review by Aiky30 — APPROVED** [review]

**Comment by Aiky30** [comment]
@ChangoFoster Glad to see you still playing with the mighty Pi mate. I hope you're well. Let me know if you're done with this and ready to merge?

**Comment by ChangoFoster** [comment]
Yep still working with it every now and again. Hope you're good too! Yeah I think I'm done with all my changes so happy to merge if you are.

**Pull request merged**